article/_posts/2022-10-07-ExplainableDisentangledRepresentations.md

```diff
@@ -0,0 +1,106 @@
+
+![](/article/images/ExplainableDisentangledRepresentations/figure1.jpg)
+
+## Autoencoder
+To learn human-interpretable factors of variation to be used later in the attribution mechanism, the authors use the
+$$\beta$$-TCVAE[^1]. Very briefly, $$\beta$$-TCVAE is a method that decomposes the KL divergence in VAEs into 3 terms,
+notably the total correlation of the aggregate posterior:
+
+$$
+KL(q(z)~\|~\prod_{j=1}^{d} q(z_{j})), \quad \text{with}~d~\text{the number of latent dimensions}
+$$
+
+which can be penalized more heavily to promote a disentangled latent space, without any additional supervision.
+
+A posteriori, the authors visually inspected traversals of individual latent dimensions to manually label the effects
+each dimension captured.
+
+## Attribution methods
+The authors mention multiple saliency methods for measuring the contribution of each $$x_i$$ dimension of the input to a
+model's predicting. Here, the authors use so called "perturbation methods", which perturb the input and measure the
+effect on the prediction, namely the expected gradients (EG)[^2] and occlusion maps (OM)[^3].
+
+The novel contribution of the paper is to introduce new attribution pathways, besides the standard pixel-wise
+*image-into-prediction* ($$A_{IP}$$) path:
+- *Latent-into-prediction* $$A_{LP}$$: Reveals what interpretable features where important for the prediction;
+- *Image-into-latent* $$A_{IL}$$: Sanity check of the method, to ensure that the captured factor of variation overlaps
+with its anticipated feature in the image.
+
+
+# Data
+The authors evaluate their method on:
+- one synthetic dataset based on MNIST (for a proof of concept);
+- OCT Retina Scans dataset[^4], containing healthy and ill patients with three diseases;
+- Skin Lesion Images[^5], containing dermoscopic images with nine different diagnostic categories.
+
+
+# Results
+The authors present a huge amount of supplementary results, but here we'll focus on the synthetic dataset and one of the
+medical imaging dataset.
+
+Their results on the synthetic dataset help understand the various attribution paths.
+
+![](/article/images/ExplainableDisentangledRepresentations/figure2.jpg)
+
+On the skin lesion dataset, the authors use their method to interpret that the skin brightness and size where most
+responsible for leading the model to mistakenly predict a mole as a melanoma.
+
+![](/article/images/ExplainableDisentangledRepresentations/figure4.jpg)
+
+This is only mentioned in the supplementary materials, but their method's interpretability/performance trade-off is
+substantial. In the interpretable settings, the best accuracies they achieved was of 95% for the synthetic dataset and
+55-60% for the OCT dataset.
+
+Using the same architecture, but allowing to finetune the encoder along with the MLP classification head (abandoning the
+latent dimensions' interpretability in the process), they were able to reach accuracies of 96% for the synthetic dataset
```

Comment:
Do they explain how to check if the accuracy gap is explained by the model's reliance on correlations?

Reply:
They don't propose a generic formulation adaptable to real world use case. They do mention that, on the toy dataset, they artificially generated some variations correlated with the prediction targets. But I find their description of these generated factors confusing, and as far as I could see they do not provide any experiments/results showing how their method can be used to improve robustness to shortcuts. They only provide one figure (in the supplementary materials) that shows the attributions on one sample when a shortcut exists.